infra/docs.sh

```diff
@@ -16,5 +16,5 @@ cp dev/docs/google2ed1af765c529f57.html dev/docs/doc
 
 # Upload the docs.
 if [ "$1" = "--upload" ]; then
-  gsutil -m rsync -d -r dev/docs/doc/ gs://docs.flutter.io/
+  gsutil -m cp -z "js,json,html,css" dev/docs/doc/ gs://docs.flutter.io/
```

Comment:
Unfortunately, cloud storage doesn't have that concept to atomically rename a directory (there isn't really directories).

We've dealt with this on other projects by putting a single file in place with a "pointer" to a new directory. Then, a server in front can read that well-known indicator to construct its path.

I'm OK with orphans in exchange for compression for now. But, we'll probably later (soon?) look for a more flexible doc solution.